pkg/operator/infrastructurecontroller/infra_controller.go

```diff
@@ -0,0 +1,79 @@
+package infrastructurecontroller
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/davecgh/go-spew/spew"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+
+	configv1client "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	operatorv1helpers "github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+)
+
+type InfrastructureController struct {
```

Comment:
name for the status field it writes.  `AWSPlatformServiceLocationController` or something similar.  Otherwise they'll write a mega controller for all infrastructure things.